sdstore/cache2disk.go

```diff
@@ -63,26 +68,32 @@ func getDirSizeInBytes(path string) int64 {
 /*
 compare md5 for source and destination directories
 param - src         		source directory
-param - dest			destination directory
+param - dest			all directory but the last element of path
+param - destBase		last element of destination directory
 return - bytearray / bool   	return md5 byte array, bool - true (md5 same) / false (md5 changed)
 */
-func checkMd5(src, dest string) ([]byte, bool) {
-	var msg string
+func checkMd5(src, dest, destBase string) ([]byte, bool) {
+	var msg, oldMd5FilePath string
 	var oldMd5 map[string]string
 	var newMd5 map[string]string
 
-	fmt.Println("start md5 check")
-	oldMd5FilePath := filepath.Join(filepath.Dir(dest), "md5.json")
+	_ = writeLog(logger.INFO, "", "start md5 check")
+	oldMd5FilePath = filepath.Join(dest, fmt.Sprintf("%s%s", destBase, ".md5"))
 	oldMd5File, err := ioutil.ReadFile(oldMd5FilePath)
 	if err != nil {
 		oldMd5File = []byte("")
-		msg = fmt.Sprintf("error: %v, not able to get md5.json from: %v", err, filepath.Dir(dest))
+		msg = fmt.Sprintf("%v, not able to get %v.md5 from: %v", err, destBase, dest)
 		_ = writeLog(logger.WARN, logger.FILE, msg)
 	}
 	_ = json.Unmarshal(oldMd5File, &oldMd5)
 
+	if newMd5, err = GenerateMd5(src); err != nil {
+		msg = fmt.Sprintf("not able to generate md5 for directory: %v", src)
+		_ = writeLog(logger.WARN, logger.MD5, msg)
+	}
+
 	if newMd5, err = MD5All(src); err != nil {
```

Comment:
this is doing md5 twice ? line 90 also seems to populate `newMd5`

Reply:
nice catch. fixed